test/functional/rpc_packages.py

```diff
@@ -49,94 +39,94 @@ def assert_testres_equal(self, package_hex, testres_expected):
         self.test_multiple_parents()
         self.test_conflicting()
         self.test_rbf()
         self.test_submitpackage()
 
-    def test_independent(self):
+    def test_independent(self, coin):
         self.log.info("Test multiple independent transactions in a package")
         node = self.nodes[0]
+
         # For independent transactions, order doesn't matter.
         self.assert_testres_equal(self.independent_txns_hex, self.independent_txns_testres)
 
         self.log.info("Test an otherwise valid package with an extra garbage tx appended")
-        garbage_tx = node.createrawtransaction([{"txid": "00" * 32, "vout": 5}], {self.address: 1})
-        tx = tx_from_hex(garbage_tx)
+        garbage_utxo = {"txid": "00" * 32, "vout": 5, "value": 3, "height": 0}
+        garbage_tx = self.wallet.create_self_transfer(utxo_to_spend=garbage_utxo)
         # Only the txid and wtxids are returned because validation is incomplete for the independent txns.
         # Package validation is atomic: if the node cannot find a UTXO for any single tx in the package,
         # it terminates immediately to avoid unnecessary, expensive signature verification.
-        package_bad = self.independent_txns_hex + [garbage_tx]
-        testres_bad = self.independent_txns_testres_blank + [{"txid": tx.rehash(), "wtxid": tx.getwtxid(), "allowed": False, "reject-reason": "missing-inputs"}]
+        package_bad = self.independent_txns_hex + [garbage_tx["hex"]]
+        testres_bad = self.independent_txns_testres_blank + [{"txid": garbage_tx["txid"], "wtxid": garbage_tx["wtxid"], "allowed": False, "reject-reason": "missing-inputs"}]
         self.assert_testres_equal(package_bad, testres_bad)
 
         self.log.info("Check testmempoolaccept tells us when some transactions completed validation successfully")
-        coin = self.coins.pop()
-        tx_bad_sig_hex = node.createrawtransaction([{"txid": coin["txid"], "vout": 0}],
-                                           {self.address : coin["amount"] - Decimal("0.0001")})
-        tx_bad_sig = tx_from_hex(tx_bad_sig_hex)
+        tx_bad_sig = self.wallet.create_self_transfer(utxo_to_spend=coin)['tx']
+        tx_bad_sig.wit.vtxinwit = []
+        tx_bad_sig_hex = tx_bad_sig.serialize().hex()
         testres_bad_sig = node.testmempoolaccept(self.independent_txns_hex + [tx_bad_sig_hex])
         # By the time the signature for the last transaction is checked, all the other transactions
         # have been fully validated, which is why the node returns full validation results for all
         # transactions here but empty results in other cases.
         assert_equal(testres_bad_sig, self.independent_txns_testres + [{
             "txid": tx_bad_sig.rehash(),
-            "wtxid": tx_bad_sig.getwtxid(), "allowed": False,
+            "wtxid": tx_bad_sig.getwtxid(),
+            "allowed": False,
             "reject-reason": "mandatory-script-verify-flag-failed (Operation not valid with the current stack size)"
```

Comment:
These changes aren't necessary, as `createrawtransaction` is not a wallet RPC and we want a tx with no signature. It's a little weird to have `MiniWallet` sign it and then manually edit the tx to remove the signature.

Reply:
Done in https://github.com/bitcoin/bitcoin/commit/17cad448516a6906ff637593ab57df332fade5d2.